client/src/Utxorpc/Logged.hs

```diff
@@ -0,0 +1,165 @@
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE RankNTypes #-}
+
+module Utxorpc.Logged
+  ( UtxorpcClientLogger,
+    RequestLogger,
+    ReplyLogger,
+    ServerChunkLogger,
+    ServerStreamEndLogger,
+    loggedSStream,
+    loggedUnary,
+  )
+where
+
+import Control.Monad.IO.Class (liftIO)
+import qualified Data.ByteString.Char8 as BS
+import Data.ProtoLens (Message)
+import Network.GRPC.Client (HeaderList, RawReply)
+import Network.GRPC.Client.Helpers (GrpcClient (..), GrpcClientConfig, UseTlsOrNot, close, rawStreamServer, rawUnary)
+import Network.GRPC.HTTP2.Encoding (GRPCInput, GRPCOutput)
+import Network.GRPC.HTTP2.Types (IsRPC (..))
+import Network.HTTP2.Client (ClientIO, HostName, PortNumber)
+import Network.HTTP2.Client.Exceptions (ClientIO)
+import Utxorpc.Types (ServerStreamReply, UnaryReply, UtxorpcService)
+
+{--------------------------------------
+  Types
+--------------------------------------}
+
+-- In the following logger types:
+--
+-- The ByteString is the method path
+--
+-- The GrcpClient is included because it contains useful information such as the server
+-- address
+--
+-- Values of type `a` are generated by the request logger and passed down the line
+-- to the other logging functions during a gRPC call.
+-- For example, the request logger performs a logging action and returns a (UUID, Int)
+-- describing the call ID and the number of replies received. For a server stream RPC,
+-- the (UUID, Int) is then passed to the server chunk logger, which increments the Int
+-- for data chunk received. The final (UUID, Int) is then passed to the ServerStreamEndLogger.
```

Comment:
I wonder if making this completely generic is a bit too flexible and if it would make the library easier to use if this type was predetermined. I assume the type is determined more by the nature of gRPC and less by the type of logging employed. However, perhaps I'll see the value of the genericity once I've seen the examples.

Reply:
I've updated the comment to explain that this type has no connection to the types of the actual RPC handlers. Hopefully the examples clear things up!

Reply:
I've given this more thought, and I think you might be right. I can't think of a use case beyond the (UUID, Into) example, and hardcoding the type would make it more understandable and easy to use.